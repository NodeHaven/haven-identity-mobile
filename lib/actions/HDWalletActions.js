// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import { STORE_ROOT_ADDRESS, INC_IDENTITY_INDEX, INC_ACCOUNT_INDEX, SEED_CONFIRMED, ACCOUNT_RISK_SENT, RESET_HD_WALLET, STORE_ALL_KEYCHAIN_ADDRESS } from 'uPortMobile/lib/constants/HDWalletActionTypes'

// These are used to keep a track of used and unused paths for a HD wallet

// ADDING NEW ACTIONS PLEASE READ THIS!!!!
//
// If the action should be backed up add the `, _backup` attribute to the action.
// When should it be backed up?
//
// - is it backed by the reducer and not a saga?
// - is it something that can not be refreshed from the block chain etc? (eg. no need to backup nonces and balances)
// - is it not just related to UX? Some reducers are temporary and would not make sense on multiple device

export function storeRootAddress (root) {
  return {
    type: STORE_ROOT_ADDRESS,
    root,
    _backup: true
  }
}

export function incIdentityIndex () {
  return {
    type: INC_IDENTITY_INDEX,
    _backup: true
  }
}

export function incAccountIndex (identity) {
  return {
    type: INC_ACCOUNT_INDEX,
    identity,
    _backup: true
  }
}

export function seedConfirmed () {
  return {
    type: SEED_CONFIRMED,
    _backup: true
  }
}

export function accountRiskSent () {
  return {
    type: ACCOUNT_RISK_SENT,
    _backup: true
  }
}

export function resetHDWallet () {
  return {
    type: RESET_HD_WALLET
  }
}


export function storeAllKeyChainAddresses (addresses) {
  return {
    type: STORE_ALL_KEYCHAIN_ADDRESS,
    addresses
  }
}