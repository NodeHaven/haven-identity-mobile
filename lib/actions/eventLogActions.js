// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import { STORE_LAST_BLOCK_FOR_LOGS, STORE_EVENTS } from '../constants/EventLogActionTypes'

// ADDING NEW ACTIONS PLEASE READ THIS!!!!
//
// If the action should be backed up add the `, _backup` attribute to the action.
// When should it be backed up?
//
// - is it backed by the reducer and not a saga?
// - is it something that can not be refreshed from the block chain etc? (eg. no need to backup nonces and balances)
// - is it not just related to UX? Some reducers are temporary and would not make sense on multiple device

const _backup = true

export function storeLastBlockForLogs (contract, block) {
  return {
    type: STORE_LAST_BLOCK_FOR_LOGS,
    contract,
    block,
    _backup
  }
}

export function storeEvents (contract, events) {
  return {
    type: STORE_EVENTS,
    contract,
    events,
    _backup
  }
}
