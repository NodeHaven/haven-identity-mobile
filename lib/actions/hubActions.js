// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import { STORE_HUB_HEAD, CATCHUP_HUB, QUEUE_EVENT, TAKE_QUEUED_EVENT, CLEAR_QUEUE, IMPORT_SNAPSHOT, RESET_HUB } from '../constants/HubActionTypes'

export function storeHubHead (hash) {
  return {
    type: STORE_HUB_HEAD,
    hash
  }
}
export function resetHub () {
  return {
    type: RESET_HUB
  }
}

export function catchupHub () {
  return {
    type: CATCHUP_HUB
  }
}

export function queueEvent (event) {
  return {
    type: QUEUE_EVENT,
    event
  }
}

export function takeQueuedEvent () {
  return {
    type: TAKE_QUEUED_EVENT
  }
}

export function clearQueue () {
  return {
    type: CLEAR_QUEUE
  }
}

export function importSnapshot (snapshot) {
  return {
    type: IMPORT_SNAPSHOT,
    snapshot
  }
}
