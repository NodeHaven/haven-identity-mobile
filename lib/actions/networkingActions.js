// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  ANALYZE_ADDRESS,
  MARK_OFFLINE,
  MARK_ONLINE,
  SAVE_GAS_PRICE
} from 'uPortMobile/lib/constants/NetworkingActionTypes'

export function goOffline () {
  return {
    type: MARK_OFFLINE
  }
}

export function goOnline () {
  return {
    type: MARK_ONLINE
  }
}

export function saveGasPrice (gasPrice) {
  return {
    type: SAVE_GAS_PRICE,
    gasPrice
  }
}

export function analyzeAddress (address) {
  return {
    type: ANALYZE_ADDRESS,
    address
  }
}
