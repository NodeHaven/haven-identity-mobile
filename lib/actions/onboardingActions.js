// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import { ADD_DATA, FETCH_CURRENT_COUNTRY, FLAG_NEWBIE, UNFLAG_NEWBIE, SET_ONBOARDING_NETWORK } from '../constants/OnboardingActionTypes'

export function addData (data) {
  return {
    type: ADD_DATA,
    data
  }
}

export function fetchCurrentCountry () {
  return {
    type: FETCH_CURRENT_COUNTRY
  }
}

export function flagNewbie () {
  return {
    type: FLAG_NEWBIE
  }
}

export function unflagNewbie () {
  return {
    type: UNFLAG_NEWBIE
  }
}

export function setOnboardingNetwork (network) {
  return {
    type: SET_ONBOARDING_NETWORK,
    network
  }
}
