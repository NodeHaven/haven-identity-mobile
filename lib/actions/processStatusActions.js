// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  SAVE_MESSAGE,
  SAVE_ERROR,
  CLEAR_MESSAGE,
  CLEAR_ALL_MESSAGES,
  START_WORKING,
  STOP_WORKING,
  COMPLETE_PROCESS,
  FAIL_PROCESS
} from 'uPortMobile/lib/constants/ProcessStatusActionTypes'

export const saveMessage = (section, message) => ({type: SAVE_MESSAGE, section, message})
export const saveError = (section, error) => ({type: SAVE_ERROR, section, error})
export const clearMessage = (section) => ({type: CLEAR_MESSAGE, section})
export const clearAllMessages = () => ({type: CLEAR_ALL_MESSAGES})
export const startWorking = (section) => ({type: START_WORKING, section})
export const stopWorking = (section) => ({type: STOP_WORKING, section})
export const completeProcess = (section) => ({type: COMPLETE_PROCESS, section})
export const failProcess = (section, error) => ({type: FAIL_PROCESS, section, error})
