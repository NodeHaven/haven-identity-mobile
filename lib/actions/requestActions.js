// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  HANDLE_URL,
  HANDLE_MESSAGE,
  SELECT_REQUEST,
  SELECT_ACCOUNT_FOR_REQUEST,
  SAVE_REQUEST,
  CLEAR_REQUEST,
  AUTHORIZE_REQUEST,
  CANCEL_REQUEST,
  CALLBACK,
  AUTHORIZE_ACCOUNT
} from 'uPortMobile/lib/constants/RequestActionTypes'

export function handleURL (url, options = {}) {
  return {
    type: HANDLE_URL,
    url,
    ...{postback: false, popup: true, clientId: null},
    ...options
  }
}

export function handleMessage (message, options = {}) {
  return {
    type: HANDLE_MESSAGE,
    message,
    ...options
  }
}

export function selectRequest (activityId) {
  return {
    type: SELECT_REQUEST,
    activityId
  }
}

// Use this in a request dialog to select the correct account
export function selectAccountForRequest (activityId, account) {
  return {
    type: SELECT_ACCOUNT_FOR_REQUEST,
    activityId,
    account
  }
}

export function saveRequest (activityId) {
  return {
    type: SAVE_REQUEST,
    activityId
  }
}

export function clearRequest () {
  return {
    type: CLEAR_REQUEST
  }
}

export function authorizeRequest (activityId) {
  return {
    type: AUTHORIZE_REQUEST,
    activityId
  }
}

export function authorizeAccount (activityId, accType) {
  return {
    type: AUTHORIZE_ACCOUNT,
    activityId: activityId,
    accType: accType
  }
}

export function cancelRequest (activityId) {
  return {
    type: CANCEL_REQUEST,
    activityId
  }
}

export function callback (request) {
  return {
    type: CALLBACK,
    request
  }
}
