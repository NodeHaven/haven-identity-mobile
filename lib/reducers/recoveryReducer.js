// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  SAVE_RECOVERY_SEED,
  HIDE_RECOVERY_SEED,
  SELECT_RECOVERY_SEED_WORD
} from 'uPortMobile/lib/constants/RecoveryActionTypes'
import {
  SWITCH_IDENTITY, RESET_IDENTITY
} from 'uPortMobile/lib/constants/UportActionTypes'

import { get, assoc, dissoc, hashMap } from 'mori'

// Requests are stored in the uport reducer. This allows you to select a request or cancel the current request
function recoveryReducer (state = hashMap(), action) {
  switch (action.type) {
    case SAVE_RECOVERY_SEED:
      return assoc(assoc(state, 'seed', action.seed), 'wordNo', 0)
    case HIDE_RECOVERY_SEED:
      return dissoc(dissoc(state, 'seed'), 'wordNo')
    case SELECT_RECOVERY_SEED_WORD:
      if (get(state, 'seed')) {
        return assoc(state, 'wordNo', action.index)
      } else {
        return state
      }
    case SWITCH_IDENTITY:
    case RESET_IDENTITY:
      return hashMap()
    default:
      return state
  }
}

export default recoveryReducer
