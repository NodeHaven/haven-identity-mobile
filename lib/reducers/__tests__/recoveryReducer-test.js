// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import recoveryReducer from 'uPortMobile/lib/reducers/recoveryReducer.js'
import {
  saveRecoverySeed,
  hideRecoverySeed,
  selectRecoveryWord
} from 'uPortMobile/lib/actions/recoveryActions'
import { switchIdentity } from 'uPortMobile/lib/actions/uportActions'

import { toJs, toClj, hashMap } from 'mori'
const empty = hashMap()
const withSeed = toClj({seed: 'SECRET SIDE', wordNo: 0})

const reducer = (state, action) => toJs(recoveryReducer(state, action))

it('saves Recovery Seed', () => {
  expect(reducer(null, saveRecoverySeed('HELLO THIS IS SECRET'))).toMatchSnapshot()
  expect(reducer(withSeed, saveRecoverySeed('HELLO THIS IS SECRET'))).toMatchSnapshot()
})

it('clears Recovery Seed', () => {
  expect(reducer(null, hideRecoverySeed())).toMatchSnapshot()
  expect(reducer(withSeed, hideRecoverySeed())).toMatchSnapshot()
})

it('select Recovery word', () => {
  expect(reducer(null, selectRecoveryWord(3))).toMatchSnapshot()
  expect(reducer(withSeed, selectRecoveryWord(3))).toMatchSnapshot()
})

it('ignores unsupported action', () => {
  expect(recoveryReducer(undefined, {type: 'UNSUPPORTED'})).toEqual(empty)
  expect(recoveryReducer(empty, {type: 'UNSUPPORTED'})).toEqual(empty)
  expect(recoveryReducer(withSeed, {type: 'UNSUPPORTED'})).toEqual(withSeed)
})

it('clears itself if main identity is switched', () => {
  expect(reducer(null, switchIdentity())).toEqual({})
  expect(reducer(withSeed, switchIdentity())).toEqual({})
})
