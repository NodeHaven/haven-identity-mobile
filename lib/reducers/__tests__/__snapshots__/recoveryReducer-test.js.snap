// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clears Recovery Seed 1`] = `null`;

exports[`clears Recovery Seed 2`] = `Object {}`;

exports[`saves Recovery Seed 1`] = `
Object {
  "seed": "HELLO THIS IS SECRET",
  "wordNo": 0,
}
`;

exports[`saves Recovery Seed 2`] = `
Object {
  "seed": "HELLO THIS IS SECRET",
  "wordNo": 0,
}
`;

exports[`select Recovery word 1`] = `null`;

exports[`select Recovery word 2`] = `
Object {
  "seed": "SECRET SIDE",
  "wordNo": 3,
}
`;
